omod/src/main/java/org/openmrs/module/labtrackingapp/page/controller/LabtrackingAddOrderPageController.java

```diff
@@ -23,6 +27,21 @@ public Object controller(PageModel model, @RequestParam(value = "appId", require
         model.addAttribute("location", uiSessionContext.getSessionLocation());
         model.addAttribute("patient", patient);
         model.addAttribute("visit", visit);
+        String zonedVisitStartDateTime = null;
+        String zonedVisitEndDateTime = null;
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+        sdf.setTimeZone(TimeZone.getDefault());
+        Calendar calendar = Calendar.getInstance();
+        if (visit != null && visit.getStartDatetime() != null) {
+            calendar.setTime(visit.getStartDatetime());
+            zonedVisitStartDateTime = sdf.format(calendar.getTime());
+            if (visit.getStopDatetime() != null) {
+                calendar.setTime(visit.getStopDatetime());
+                zonedVisitEndDateTime = sdf.format(calendar.getTime());
+            }
+        }
+        model.addAttribute("visitStartDateTime", zonedVisitStartDateTime);
+        model.addAttribute("visitEndDateTime", zonedVisitEndDateTime);
```

Comment:
Interesting, I would think that the server would automatically be including the time zone? did we confirm this is neeed?

Reply:
@mogoodrich , please see my comment below.

Reply:
Adding the visitStartDateTime in the controller was just a quick around to get this working. Another option would be that on the client side to use the visit REST API to obtain the visit info, and that would include all the datetimes properties with the timezone info.